projects/step-core/src/lib/components/import-dialog/import-dialog.component.ts

```diff
@@ -0,0 +1,60 @@
+import { Observable, tap } from 'rxjs';
+import { Component, HostListener, inject } from '@angular/core';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+import { AlertType, DialogsService, ImportsService } from '@exense/step-core';
+import { ImportDialogData } from '../../shared/import-dialog-data.interface';
+
+@Component({
+  selector: 'step-plan-import-dialog',
+  templateUrl: './import-dialog.component.html',
+  styleUrls: ['./import-dialog.component.scss'],
+})
+export class ImportDialogComponent {
+  private _matDialogRef = inject<MatDialogRef<ImportDialogComponent, boolean>>(MatDialogRef);
+  private _dialogs = inject(DialogsService);
+  private _data = inject<ImportDialogData>(MAT_DIALOG_DATA);
+  private _importsService = inject(ImportsService);
+
+  protected showOptions = true;
+  protected isImporting = false;
+  protected resourcePath!: string;
+  protected importAll = this._data.importAll;
+  protected overwrite = this._data.overwrite;
+
+  readonly title = this._data.title;
+  readonly entity = this._data.entity;
+  readonly AlertType = AlertType;
+
+  @HostListener('keydown.enter')
+  save(): void {
+    if (this.resourcePath) {
+      this.invokeImport()
+        .pipe(
+          tap(() => {
+            this.isImporting = true;
+          })
+        )
+        .subscribe({
+          next: (response: any) => {
+            this._matDialogRef.close();
+
+            if (response && response.length > 0) {
+              this._dialogs.showListOfMsgs(response);
+            }
+          },
+          complete: () => {
+            this.isImporting = false;
+          },
+          error: (error: any) => console.error(error),
+        });
+    } else {
+      this._dialogs.showErrorMsg('Upload not completed.');
+    }
```

Comment:
It will be easier to read, in case if the amount of nested blocks will be reduced.
Also try to avoid `any` usage

```suggestion
    if (!this.resourcePath) {
        this._dialogs.showErrorMsg('Upload not completed.');       
        return;
    }
 
    this.invokeImport()
      .pipe(
        tap(() => {
          this.isImporting = true;
        })
      )
      .subscribe({
        next: (response: string[]) => {
          this._matDialogRef.close();
           if (!!response?.length) {
            this._dialogs.showListOfMsgs(response);
          }
        },
        complete: () => {
          this.isImporting = false;
        },
        error: (error: unknown) => console.error(error),
       });
```

Reply:
thanks!